src/Components/v2/CarouselV2.tsx

```diff
@@ -0,0 +1,265 @@
+import { Box, ChevronIcon, Flex } from "@artsy/palette"
+import React, { ReactNode, useEffect, useRef, useState } from "react"
+import styled from "styled-components"
+import { left, LeftProps, right, RightProps } from "styled-system"
+import { Media } from "Utils/Responsive"
+
+export interface FlickitySettings {
+  draggable?: boolean
+  freeScroll?: boolean
+  wrapAround?: boolean
+  groupCells?: boolean | number | string
+  autoPlay?: boolean | number
+  fullscreen?: boolean
+  fade?: boolean
+  adaptiveHeight?: boolean
+  watchCSS?: boolean
+  hash?: boolean
+  dragThreshold?: number
+  friction?: number
+  selectedAttraction?: number
+  freeScrollFriction?: number
+  imagesLoaded?: boolean
+  lazyLoad?: boolean | number
+  bgLazyLoad?: boolean | number
+  initialIndex?: number
+  accessibility?: boolean
+  setGallerySize?: boolean
+  resize?: boolean
+  cellAlign?: "left" | "center" | "right"
+  contain?: boolean
+  percentPosition?: boolean
+  rightToLeft?: boolean
+  prevNextButtons?: boolean
+  pageDots?: boolean
+}
+
+type Arrow = (
+  props: {
+    Arrow: React.ReactType
+    getFlickity: any
+  } & Props
+) => React.ReactNode
+
+interface Props {
+  height?: number
+  data: object[] // This is designed to handle any shape of data passed, as long as its an array
+  render: (slide) => ReactNode
+  renderLeftArrow?: Arrow
+  renderRightArrow?: Arrow
+  onArrowClick?: () => void
+  settings?: FlickitySettings
+}
+
+export class Carousel extends React.Component<Props> {
+  static defaultProps = {
+    height: 300,
+  }
+
+  render() {
+    return (
+      <>
+        <Media at="xs">
+          <SmallCarousel {...this.props} />
+        </Media>
+        <Media greaterThan="xs">
+          <LargeCarousel {...this.props} />
+        </Media>
+      </>
+    )
+  }
+}
+
+/*
+ * Returns null if no window for SSR
+ */
+const Flickity =
+  typeof window !== "undefined"
+    ? require("react-flickity-component")
+    : () => null
+
+export const LargeCarousel = (props: Props) => {
+  const [isMounted, toggleMounted] = useState(false)
+  const [currentSlideIndex, setSlideIndex] = useState(0)
+  const [lastItemVisible, setLastItemVisible] = useState(false)
+
+  useEffect(() => {
+    toggleMounted(true)
+  }, [])
```

Comment:
Since this is a somewhat common requirement, it might be good to move this to a custom hook `Utils/hooks/useDidMount.ts`. Something like:

```js
export function useDidMount() {
  const [isMounted, toggleMounted] = useState(false)

  useEffect(() => {
    toggleMounted(true)
  }, [])

  return isMounted
}
```

And then later, in this carousel, can do 

```
const isMounted = useDidMount()
```

It would also be good to announce in the #front-end that we now have a folder under `Utils/hooks` where common hooks are stored and that folks should contribute there.

Reply:
Do you think each hook should have it's own file or just one file for all hooks...`hooks.ts`

Reply:
Own file -- small files are ok :) That way devs can scan their folder directory to see whats available, vs having to dig into files.